UT: ZOO-NB-UnitTests-2.3-PYTHON test failed

org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 19.0 failed 1 times, most recent failure: Lost task 0.0 in stage 19.0 (TID 10, localhost, executor driver): java.lang.ArithmeticException: / by zero

jenkins link:

http://10.239.47.210:18888/job/ZOO-NB-UnitTests-2.3-PYTHON/909/consoleFull

http://10.239.47.210:18888/job/ZOO-NB-UnitTests-2.1-PYTHON/1138/console

```
>       estimator.fit(df)

../test/zoo/pipeline/nnframes/test_nn_classifier.py:209: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/work/spark-2.1.3/python/lib/pyspark.zip/pyspark/ml/base.py:64: in fit
    return self._fit(dataset)
/opt/work/spark-2.1.3/python/lib/pyspark.zip/pyspark/ml/wrapper.py:236: in _fit
    java_model = self._fit_java(dataset)
/opt/work/spark-2.1.3/python/lib/pyspark.zip/pyspark/ml/wrapper.py:233: in _fit_java
    return self._java_obj.fit(dataset._jdf)
/opt/work/spark-2.1.3/python/lib/py4j-0.10.7-src.zip/py4j/java_gateway.py:1257: in __call__
    answer, self.gateway_client, self.target_id, self.name)
/opt/work/spark-2.1.3/python/lib/pyspark.zip/pyspark/sql/utils.py:63: in deco
    return f(*a, **kw)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

answer = 'xro20402'
gateway_client = <py4j.java_gateway.GatewayClient object at 0x7f98d1332550>
target_id = 'o20367', name = 'fit'

    def get_return_value(answer, gateway_client, target_id=None, name=None):
        """Converts an answer received from the Java gateway into a Python object.
    
        For example, string representation of integers are converted to Python
        integer, string representation of objects are converted to JavaObject
        instances, etc.
    
        :param answer: the string returned by the Java gateway
        :param gateway_client: the gateway client used to communicate with the Java
            Gateway. Only necessary if the answer is a reference (e.g., object,
            list, map)
        :param target_id: the name of the object from which the answer comes from
            (e.g., *object1* in `object1.hello()`). Optional.
        :param name: the name of the member from which the answer comes from
            (e.g., *hello* in `object1.hello()`). Optional.
        """
        if is_error(answer)[0]:
            if len(answer) > 1:
                type = answer[1]
                value = OUTPUT_CONVERTER[type](answer[2:], gateway_client)
                if answer[1] == REFERENCE_TYPE:
                    raise Py4JJavaError(
                        "An error occurred while calling {0}{1}{2}.\n".
>                       format(target_id, ".", name), value)
E                   py4j.protocol.Py4JJavaError: An error occurred while calling o20367.fit.
E                   : java.lang.UnsupportedOperationException: empty collection
E                   	at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$apply$36.apply(RDD.scala:1028)
E                   	at org.apache.spark.rdd.RDD$$anonfun$reduce$1$$anonfun$apply$36.apply(RDD.scala:1028)
E                   	at scala.Option.getOrElse(Option.scala:121)
E                   	at org.apache.spark.rdd.RDD$$anonfun$reduce$1.apply(RDD.scala:1028)
E                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
E                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
E                   	at org.apache.spark.rdd.RDD.withScope(RDD.scala:362)
E                   	at org.apache.spark.rdd.RDD.reduce(RDD.scala:1008)
E                   	at com.intel.analytics.bigdl.optim.DistriOptimizer$.optimize(DistriOptimizer.scala:162)
E                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
E                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
E                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
E                   	at java.lang.reflect.Method.invoke(Method.java:498)
E                   	at com.intel.analytics.zoo.pipeline.api.keras.layers.utils.KerasUtils$.invokeMethod(KerasUtils.scala:302)
E                   	at com.intel.analytics.zoo.pipeline.api.keras.layers.utils.KerasUtils$.invokeMethodWithEv(KerasUtils.scala:329)
E                   	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalOptimizerUtil$.optimizeModels(Topology.scala:1063)
E                   	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalDistriOptimizer.train(Topology.scala:1262)
E                   	at com.intel.analytics.zoo.pipeline.api.keras.models.InternalDistriOptimizer.train(Topology.scala:1472)
E                   	at com.intel.analytics.zoo.pipeline.nnframes.NNEstimator.internalFit(NNEstimator.scala:470)
E                   	at com.intel.analytics.zoo.pipeline.nnframes.NNEstimator.internalFit(NNEstimator.scala:198)
E                   	at org.apache.spark.ml.DLEstimatorBase.fit(DLEstimatorBase.scala:58)
E                   	at org.apache.spark.ml.DLEstimatorBase.fit(DLEstimatorBase.scala:50)
E                   	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
E                   	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
E                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
E                   	at java.lang.reflect.Method.invoke(Method.java:498)
E                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
E                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
E                   	at py4j.Gateway.invoke(Gateway.java:282)
E                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
E                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)
E                   	at py4j.GatewayConnection.run(GatewayConnection.java:238)
E                   	at java.lang.Thread.run(Thread.java:748)
```

--- a/pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py
+++ b/pyzoo/test/zoo/pipeline/nnframes/test_nn_classifier.py
@@ -200,7 +200,7 @@ class TestNNClassifer():
             ((2.0, 1.0), (1.0, 2.0)),
             ((1.0, 2.0), (2.0, 1.0)),
             ((2.0, 1.0), (1.0, 2.0)),
-            ((1.0, 2.0), (2.0, 1.0))])
+            ((1.0, 2.0), (2.0, 1.0))] * 10)
 
         schema = StructType([
             StructField("features", ArrayType(DoubleType(), False), False),

